ee/clickhouse/views/test/test_clickhouse_retention.py

```diff
@@ -1,32 +1,143 @@
 import json
 import numbers
-from typing import List, Literal, TypedDict, Union
+from dataclasses import asdict, dataclass
+from typing import List, Literal, Optional, TypedDict, Union
 
 from django.test import TestCase
 from django.test.client import Client
 
-from ee.clickhouse.test.test_journeys import journeys_for
+from ee.clickhouse.test.test_journeys import _create_all_events, update_or_create_person
 from ee.clickhouse.views.test.funnel.util import EventPattern
+
+# from posthog.api.test.test_event import get_events_ok
 from posthog.api.test.test_organization import create_organization
 from posthog.api.test.test_team import create_team
 from posthog.api.test.test_user import create_user
 
 
 class RetentionTests(TestCase):
-    def test_can_get_line_chart_and_fetch_people(self):
+    def test_can_get_retention_cohort_breakdown(self):
+        organization = create_organization(name="test")
+        team = create_team(organization=organization)
+        user = create_user(email="test@posthog.com", password="1234", organization=organization)
+
+        self.client.force_login(user)
+
+        update_or_create_person(distinct_ids=["person 1"], team_id=team.pk)
+        update_or_create_person(distinct_ids=["person 2"], team_id=team.pk)
+        update_or_create_person(distinct_ids=["person 3"], team_id=team.pk)
+
+        setup_user_activity_by_day(
+            daily_activity={
+                "2020-01-01": {"person 1": [{"event": "target event"}], "person 2": [{"event": "target event"}]},
+                "2020-01-02": {"person 1": [{"event": "target event"}], "person 3": [{"event": "target event"}]},
+                "2020-01-03": {"person 1": [{"event": "target event"}], "person 3": [{"event": "target event"}]},
+            },
+            team=team,
+        )
+
+        retention = get_retention_ok(
+            client=self.client,
+            team_id=team.pk,
+            request=RetentionRequest(
+                target_entity={"id": "target event", "type": "events"},
+                returning_entity={"id": "target event", "type": "events"},
+                date_from="2020-01-01",
+                total_intervals=2,
+                date_to="2020-01-02",
+                period="Day",
+                retention_type="retention_first_time",
+            ),
+        )
+
+        retention_by_cohort_by_period = get_by_cohort_by_period_from_response(response=retention)
+
+        assert retention_by_cohort_by_period == {
+            "Day 0": {"1": 2, "2": 1,},  # ["person 1", "person 2"]  # ["person 1"]
+            "Day 1": {"1": 1},  # ["person 3"]
+        }
```

Comment:
Is this the breakdown-less case now?

Reply:
Yes, I just added this to ensure that the existing functionality stays the same

Reply:
The test name is confusing, as it's `test_can_get_retention_cohort_breakdown`, so I wasn't sure if this _should_ be including a breakdown or not.
Beyond the test itself, from a user's perspective the breakdown-ful result type seems to pivot quite a bit from the breakdown-less one. E.g. I'd be asking myself: Which day (0, 1, 2, 3?) am I seeing when I do a breakdown by properties? Why can't I continue to see all days with a property breakdown? This is also partly a UI concern – maybe the best solution is just showing a default breakdown state (instead of _just_ "Add breakdown") that would say "Following week". But then it would still be a bit confusing that there are seemingly two modes to the Retention insight. Am I missing something?

Reply:
Good point. On the naming, I was thinking about the traditional non-breakdown case as a special case of a breakdown, e.g. here your `breakdown_values` is essentially the interval of the first event. But perhaps that's not an intuitive view of it?

On the breakdown and "which day am I seeing" would you mind expanding on what you would expect to see? Maybe a hangout would be helpful if you have the time?

Reply:
Ah, for me the default case is not so much an implicit breakdown by the interval, but more like trends: a time series – which is orthogonal to the breakdown.

I guess the thing I'd expect to see having seen the default case first is to still have the multiple intervals, each being broken down. This way it would always be a time series. Or we can ditch the time series and in the default case just return one row – I'm not sure which is best, as I don't have the best context on what exactly our Retention users are looking to discover

Reply:
@Twixes I'm going to follow up with an addition of the cohort breakdowns under the top level breakdowns. This appears to be how e.g. Amplitude handles it, not that we should take that as gospel.

For the time series line graph we will get something like [this](https://github.com/PostHog/posthog/pull/7340) with each line being a breakdown value.